src/common/config_opts.h

```diff
@@ -899,6 +899,7 @@ OPTION(osd_min_pg_log_entries, OPT_U32, 3000)  // number of entries to keep in t
 OPTION(osd_max_pg_log_entries, OPT_U32, 10000) // max entries, say when degraded, before we trim
 OPTION(osd_force_recovery_pg_log_entries_factor, OPT_FLOAT, 1.3) // max entries factor before force recovery
 OPTION(osd_pg_log_trim_min, OPT_U32, 100)
+OPTION(osd_pg_log_dup_track_versions, OPT_U32, 1000) // how many versions back to track operations that could be duplicated in the pg log
```

Comment:
This is currently trailing dup entries past the end of the pg log. I like the idea of (3) - changing this option to be total dup + pg log entries, so it adjusts the dup size based on the pg log size. Then we could make it 3000 (the current min pg log length) and lower the default min pg log length to e.g. 1000 to save memory when not recovering/backfilling/degraded.

(1) is taken care of already by calc_trim_to() - the pg log is trimmed less (only to the max, 10000 by default) during recovery/degraded/backfill.